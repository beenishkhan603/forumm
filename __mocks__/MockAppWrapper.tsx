import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
