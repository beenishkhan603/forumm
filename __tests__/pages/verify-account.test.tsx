import { render } from '@testing-library/react'
import LoginPage from 'pages/login'
import VerifyAccount from 'pages/verify-account'
import MockAppWrapper from '__mocks__/MockAppWrapper'

it('renders verify account unchanged', () => {
  const { container } = render(
    <MockAppWrapper>
      <VerifyAccount />
    </MockAppWrapper>
  )
  expect(container).toMatchSnapshot()
})
