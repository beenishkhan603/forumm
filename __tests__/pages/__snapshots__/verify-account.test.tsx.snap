// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders verify account unchanged 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="flex justify-center"
    style="color: rgb(24, 26, 32);"
  >
    <div
      class="w-full min-h-screen"
      style="background-color: rgb(255, 255, 255); color: rgb(24, 26, 32);"
    >
      <div
        class="w-full mx-auto md:w-1/2 my-16 p-10"
        style="background-color: rgb(255, 255, 255); color: rgb(24, 26, 32);"
      >
        <div
          class="text-white text-center text-2xl font-bold pl-3 md:pl-0 mb-8"
          style="color: rgb(24, 26, 32);"
        >
          Welcome to Forumm
          <svg
            class="inline-block w-12"
          />
        </div>
        <div
          class="w-full flex justify-center"
          style="color: rgb(24, 26, 32);"
        >
          <div
            class="w-32 h-32 "
            data-testid="loading"
            style="color: rgb(24, 26, 32);"
          >
            <svg
              class="w-32 h-32 "
            />
            <svg
              class="w-32 h-32  !h-12"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
